### PR TITLE
Use a process controller to handle shipping market refreshes

### PR DESCRIPTION
**Pull request by pgmzeta**
[qol][bug][station systems]
## About the PR 
Move shipping market refresh handling to a process controller.

Note: This will cause mail deliveries to occur even when there are no QMs/Mail Couriers.

## Why's this needed? 
Mail couriers do not have QM console access, and thus cannot recieve more mail until the market has refreshed. If there is no active QMs for whatever reason, then the mail courier can be SOL with their one job.
Stifle some small cheese where you can keep hot items hot indefinitely if no one ever checks the console.

## Changelog 

```changelog
(u)glowbold
(*)Market refreshes and mail deliveries will now occur automatically.
```

### CODE REVIEW

**Review by frawhst — CHANGES_REQUESTED** [review]
This should definitely be a process loop instead of a spawn loop. See `code/datums/controllers/process/stock_market.dm` or `code/datums/controllers/process/ticker.dm` for simple examples of what those look like. Also, we probably don't need the code to manually refresh the market since it's handled automatically, so feel free to axe that.

File: code/modules/economy/shippingmarket.dm [hunk 1]
```diff
@@ -72,9 +71,6 @@
 				qdel(C)
 
 
-		SPAWN(300)
-			market_shift()
-
```
Comment (Tarmunora) [inline code/modules/economy/shippingmarket.dm:77]: hm
without this early market shift, when does the first mail crate actually arrive in a round? at the ~7 minute mark? ~14?

Comment (pgmzeta) [reply code/modules/economy/shippingmarket.dm:77]: It comes in at ~7 since the first market shift now happens at round-start

**Review by frawhst — APPROVED** [review]

File: code/modules/economy/shippingmarket.dm [hunk 2]
```diff
@@ -151,27 +144,17 @@
...
+	/// Returns the time in MM:SS format
 	proc/get_market_timeleft()
 		var/timeleft = src.timeleft() / 10
 		if(timeleft)
 			return "[add_zero(num2text((timeleft / 60) % 60),2)]:[add_zero(num2text(timeleft % 60), 2)]"
 
 	proc/market_shift()
-		#ifndef FUCK_OFF_WITH_THE_MAIL
-		var/time_since_previous = (TIME - last_market_update)
-		#endif
```
[the remark this answers is not in the document]
Comment (pgmzeta) [reply code/modules/economy/shippingmarket.dm:172]: still used down here

https://github.com/goonstation/goonstation/blob/febafea5807f83bc2ed355610874e6f169e8cd7b/code/modules/economy/shippingmarket.dm#L280-L284

**Pull request merged**